### PR TITLE
Game and Author implementations

### PR DESCRIPTION
**Pull request by teckim**
### In this pull request:
- Create a Game class in a separate .rb file.
- Create an Author class with an association to the Item class (in a separate .rb file).
- All Game class properties visible in the diagram should be defined and set up in the constructor method.
- All Author class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Author class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Game class
        - should override the method from the parent class
        - should return true if parent's method returns true AND if last_played_at is older than 2 years
        - otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options should be available:
    - List of games
    - List all authors (e.g. 'Stephen King')
    - Add a game
 - All data should be preserved by saving collections in .json files.
 - Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
    - games table (add all properties and associations from the parent Item class as table columns)
    - authors table

### CODE REVIEW

File: author.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  def initialize(first_name:, last_name:, id: Random.rand(1..1000), items: [])
+    @id = id
+    @first_name = first_name
+    @last_name = last_name
+    @items = items
+  end
+
+  def add_item(item)
+    @items << item unless @items.include? item
+  end
```
Comment (sntakirutimana72) [inline author.rb]: Kindly, implement all required behaviors for the `add_item` method. Assuming the parameter `item` has not set its `author` attribute, it should do so here as per requirement.
> - should add self as a property of the item object (by using the correct setter from the item object).

**Review by sntakirutimana72 — CHANGES_REQUESTED** [review]
Hello @teckim

You have done a tremendous job so far. However, there are some key points that need improvement.

## Highlights

✔️ You followed Ruby's best practice
✔️ Codebase is well organized

## REQUIRED CHANGES

Kindly,
  - fix linter errors
  - fix your test cases.
  - fix your app exit strategy as it fails when you attempt to quit.
  - create a `schema.sql` and add appropriate content as per project requirements.
  - refer to the comments below for other areas to improve.

## [Optional] Changes

N/A

------

Feel free to reach out if you require clarification on anything. Remember to tag me in your comments if I'm to respond.

Cheers & happy coding!! 🥂 💻

File: author.rb (hunk 1 above)
Comment (teckim) [reply author.rb]: Thank you @sntakirutimana72, I missed that

**Comment by teckim** [comment]
> Hello @teckim
> 
> You have done a tremendous job so far. However, there are some key points that need improvement.
> 
> ## Highlights
> ✔️ You followed Ruby's best practice ✔️ Codebase is well organized
> 
> ## REQUIRED CHANGES
> Kindly,
> 
> * fix linter errors
> * fix your test cases.
> * fix your app exit strategy as it fails when you attempt to quit.
> * create a `schema.sql` and add appropriate content as per project requirements.
> * refer to the comments below for other areas to improve.
> 
> ## [Optional] Changes
> N/A
> 
> Feel free to reach out if you require clarification on anything. Remember to tag me in your comments if I'm to respond.
> 
> Cheers & happy coding!! 🥂 💻

Thank you @sntakirutimana72 for your valuable review. Can you please explain more about what you mean by this point: fix your app exit strategy as it fails when you attempt to quit.

Also, I think there are some improvements related to Ali's work so I couldn't work on it.

**Pull request merged**